### ISSUE TITLE
belongsTo tracking not working under Ember Data 3.5.0

### ISSUE THREAD

**Issue by mwpastore** [danielspaniel/ember-data-change-tracker#54]

**Comment by ptgamr** [comment]
Running into this as well...  Don't know what model-framents is ... do you know when it will be fix in ember-data ?

**Comment by mwpastore** [comment]
@ptgamr https://github.com/lytics/ember-data-model-fragments

**Comment by danielspaniel** [comment]
model fragments people are agonizingly slow to update .. of course we could just do the updates and fixes ourselves with PR, but someone has already gotten close ( so i am just sitting by the fire reading with a super long book hoping they finish by the time that book is over )

**Comment by ptgamr** [comment]
@danielspaniel thanks! any idea how to make it work in the mean time? 

I take a look in to the tracker.js code, it looks like the `serialize(model, key)` return undefined for `async belongsTo` ? Could that be if we return the id, then that would be ok?

**Comment by ptgamr** [comment]
This line: https://github.com/danielspaniel/ember-data-change-tracker/blob/1186b1a79a790d3879c8c79c526351113df2571e/addon/tracker.js#L406

**Comment by danielspaniel** [comment]
you could make a fork of change-tracker @ptgamr   , fix that line and see if that does it  .. and if it does then when the fragments thing is fixed submit that with a PR and I will merge it into change tracker .. ( I am guessing it will only fail when we update to 3.5 so we can wait till then. )

**Comment by mwpastore** [comment]
@ptgamr FWIW, I've been using Ember Data 3.4.4, which has 99% of what I need (plus a squirrelly bug or two, unfortunately) until there's progress on this.

**Comment by danielspaniel** [comment]
I guess we would have to remove the model fragments dependency. I vote for that one since fragments people are not maintaining anymore. you have time for such a thing?

**Comment by danielspaniel** [comment]
though then we have to fix the belongsTo issue after that ( I guess? )

**Comment by BnitoBzh** [comment]
I did not take the time to look at which part of the code use the EDMF addon.
But are not you the best person to deal with this point?

**Comment by danielspaniel** [comment]
yes, maybe ?  i guess anyone can do that. sure, i can do faster perhaps, but sometimes people want to contribute to the fun and that sort of thing.

**Comment by SergeAstapov** [comment]
@danielspaniel FYI [ember-data-model-fragments](https://github.com/lytics/ember-data-model-fragments/blob/master/CHANGELOG.md#v400-january-25-2018) v4 was released.

**Comment by danielspaniel** [comment]
Yes, I saw that @SergeAstapov , and I already tried to fix change tracker and there were many test fails with ED 3.5 .. would you have any time to give it try ( to fix stuff ?? )  I have new version of factory-guy 3.6  so you will be able to see the test fails. i am not using ember so much anymore and am not trying to be a poopy pants quitter, because I love this addon, but if you are still rocking with ember data then this is good chance to help out.  This addon uses all public api and there is no trickery going on.

**Comment by danielspaniel** [comment]
never mind @SergeAstapov , i think i got it
